### PR TITLE
Toggle container when amenity is selected

### PR DESCRIPTION
**Pull request by prestonp**
For #28 I just made the container slide away when an amenity is selected.

### CODE REVIEW

File: client/js/components/Container.jsx [hunk 1]
```diff
@@ -17,13 +17,19 @@ export default class Container extends React.Component {
     }
 
     render() {
+        // Pass props to dynamically rendered children
+        var props = { slideUp: this.slideUp.bind(this) };
+        const children = React.Children.map(this.props.children, (element) =>
+            React.cloneElement(element, props)
+        );
```
Comment (luqmaan) [inline client/js/components/Container.jsx:24]: Would it be cleaner to use `...` to transfer the props? https://facebook.github.io/react/docs/transferring-props.html

File: client/js/components/Container.jsx [hunk 2]
```diff
@@ -17,13 +17,19 @@ export default class Container extends React.Component {
     }
 
     render() {
+        // Pass props to dynamically rendered children
+        var props = { slideUp: this.slideUp.bind(this) };
```
Comment (luqmaan) [inline client/js/components/Container.jsx]: Don't need to bind. :dancer: 

We have fat arrows.

```
{slideUp: () => this.slideUp()}
```

File: client/js/components/ParkFeatureListItem.jsx [hunk 3]
```diff
@@ -4,18 +4,23 @@ import icons from '../utils/icons.json';
 
 
 export default class ParkFeatureListItem extends React.Component {
+    onClick() {
+        this.props.selectFeature(this.props.id);
+        this.props.slideUp();
```
Comment (luqmaan) [inline client/js/components/ParkFeatureListItem.jsx:9]: I think the call to `this.props.slideUp()` should live in `selectFeature()`.

That way `ParkFeatureListItem` doesn't need to know about `slideUp()`.

File: client/js/components/Container.jsx (hunk 1 above)
Comment (prestonp) [inline client/js/components/Container.jsx:24]: A few lines down, the children are rendered using `{this.props.children}`

I wasn't sure how to add other props into the children

Comment (luqmaan) [inline client/js/components/Container.jsx:24]: :+1:

File: client/js/components/ParkFeatureListItem.jsx (hunk 3 above)
Comment (prestonp) [inline client/js/components/ParkFeatureListItem.jsx:9]: Hmm, so I was initially going to do that but 
- ParkPage defines selectFeature
- Container defines slideUp
- Container is a subcomponent of ParkPage

So the parent would have to reach into its child's properties in order to invoke slideUp. I think this is anti-pattern for react?

File: client/js/components/Container.jsx (hunk 2 above)
Comment (prestonp) [inline client/js/components/Container.jsx]: Cool, I'll change to an arrow fn

File: client/js/components/ParkFeatureListItem.jsx (hunk 3 above)
Comment (luqmaan) [inline client/js/components/ParkFeatureListItem.jsx:9]: This is where the action creators from flux become nice. :smiley_cat:

**Comment by luqmaan** [comment]
:deciduous_tree:

Good point about ParkPage being the parent of Container. We may end up needing to consider flux or something like that at some point.

File: client/js/components/ParkFeatureListItem.jsx (hunk 3 above)
Comment (prestonp) [inline client/js/components/ParkFeatureListItem.jsx:9]: oo I haven't looked into flux, I'll read up on it :book:

Comment (aaronmgdr) [inline client/js/components/ParkFeatureListItem.jsx:9]: i noticed this slide/up/down thing when doing the routing. currently there wasnt a simple way i could see if having the router router to the up state. something to think about

**Comment by luqmaan** [comment]
@prestonp you should have the permissions to merge this pr yourself btw

**Comment by prestonp** [comment]
cool, wasn't sure if you wanted to sign off on PRs or not.

**Comment by luqmaan** [comment]
:deciduous_tree: = sign off

**Pull request merged**